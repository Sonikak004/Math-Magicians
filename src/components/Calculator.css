.calculator {
  width: 100%;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.keypad {
  width: 40%;
  display: grid;
  grid-template-rows: repeat(6, 1fr);
}

.quote {
  width: 51%;
  text-align: center;
  background: #f5913e;
  color: #fff;
  margin-bottom: 20px;
  border-radius: 9px 9px 9px 9px;
}

.row {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

.row:first-child {
  grid-template-columns: 1fr;
}

.row:last-child {
  grid-template-columns: 2fr 1fr 1fr;
}

.display {
  text-align: right;
  padding: 10px;
  background-color: #858693;
  color: white;
  border: none;
}

.it-color {
  background-color: #f5913e;
  border: 1px solid #b9b7b7;
}

.item {
  border: 1px solid #d1d1d1;
}

.item:hover {
  background-color: #858693;
}
