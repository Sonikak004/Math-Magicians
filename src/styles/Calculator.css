.calculator {
  width: 100%;
  display: flex;
  align-items: flex-end;
  justify-content: center;
  flex-direction: column;
  position: relative;
  right: 30px;
  height: 2vh;
}

.keypad {
  width: 40%;
  top: 47px;
  padding: 5px;
  display: grid;
  border: 1px solid black;
  grid-template-rows: repeat(6, 1fr);
  position: relative;
}

.row {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

.row:first-child {
  grid-template-columns: 1fr;
}

.row:last-child {
  grid-template-columns: 2fr 1fr 1fr;
}

.display {
  text-align: right;
  padding: 10px;
  background-color: #a95521;
  color: white;
  border: none;
}

.it-color {
  background-color: #1176d5;
  border: 1px solid #b9b7b7;
}

.item {
  border: 1px solid #d1d1d1;
}

.item:hover {
  background-color: #858693;
}

.calc-heading {
  width: 94.5%;
  top: 46px;
  position: relative;
  color: #a575e3;
  font-size: 30px;
}
