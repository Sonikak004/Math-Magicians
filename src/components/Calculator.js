import './Calculator.css';
import React from 'react';
import ButtonRow from './rowButton';

function Calculator() {
  return (
    <div className="calculator">
      <div className="display">0</div>
      <ButtonRow />
    </div>
  );
}

export default Calculator;
