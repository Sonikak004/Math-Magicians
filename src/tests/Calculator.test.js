import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

import Calculator from '../components/Calculator';

describe('Calculator component', () => {
  test('renders correctly', () => {
    render(<Calculator />);
    const calculatorElement = screen.getByText(/Lets do some Math!/i);
    expect(calculatorElement).toBeInTheDocument();
  });

  test('calculates addition correctly', () => {
    render(<Calculator />);
    const displayElement = screen.getByText('0', { selector: '.display' });

    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('='));

    expect(displayElement).toHaveTextContent('12');
  });
  test('calculates subtraction correctly', () => {
    render(<Calculator />);
    const displayElement = screen.getByText('0', { selector: '.display' });

    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('-'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('='));

    expect(displayElement).toHaveTextContent('2');
  });
  test('calculates division correctly', () => {
    render(<Calculator />);
    const displayElement = screen.getByText('0', { selector: '.display' });

    fireEvent.click(screen.getByText('8'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('='));

    expect(displayElement).toHaveTextContent('16');
  });
  test('matches snapshot', () => {
    const { asFragment } = render(<Calculator />);
    expect(asFragment()).toMatchSnapshot();
  });
});
