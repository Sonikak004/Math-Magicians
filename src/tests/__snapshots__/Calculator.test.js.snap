// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="calculator"
  >
    <h2
      class="calc-heading"
    >
       Lets do some Math!
    </h2>
    <div
      class="keypad"
    >
      <div
        class="row"
      >
        <div
          class="display"
          data-testid="display"
        >
          0
        </div>
      </div>
      <div
        class="row"
      >
        <button
          class="item"
          name="AC"
          type="button"
        >
          AC
        </button>
        <button
          class="item"
          name="+/-"
          type="button"
        >
          +/-
        </button>
        <button
          class="item"
          name="%"
          type="button"
        >
          %
        </button>
        <button
          class="item it-color"
          name="รท"
          type="button"
        >
          รท
        </button>
      </div>
      <div
        class="row"
      >
        <button
          class="item"
          name="7"
          type="button"
        >
          7
        </button>
        <button
          class="item"
          name="8"
          type="button"
        >
          8
        </button>
        <button
          class="item"
          name="9"
          type="button"
        >
          9
        </button>
        <button
          class="item it-color"
          name="x"
          type="button"
        >
          x
        </button>
      </div>
      <div
        class="row"
      >
        <button
          class="item"
          name="4"
          type="button"
        >
          4
        </button>
        <button
          class="item"
          name="5"
          type="button"
        >
          5
        </button>
        <button
          class="item"
          name="6"
          type="button"
        >
          6
        </button>
        <button
          class="item it-color"
          name="-"
          type="button"
        >
          -
        </button>
      </div>
      <div
        class="row"
      >
        <button
          class="item"
          name="1"
          type="button"
        >
          1
        </button>
        <button
          class="item"
          name="2"
          type="button"
        >
          2
        </button>
        <button
          class="item"
          name="3"
          type="button"
        >
          3
        </button>
        <button
          class="item it-color"
          name="+"
          type="button"
        >
          +
        </button>
      </div>
      <div
        class="row"
      >
        <button
          class="item"
          name="0"
          type="button"
        >
          0
        </button>
        <button
          class="item"
          name="."
          type="button"
        >
          .
        </button>
        <button
          class="item it-color"
          name="="
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
